### PR TITLE
Port fundamentals

### PR DESCRIPTION
**Pull request by ben-challis**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            php-image: "public.ecr.aws/docker/library/php:8.2-cli-alpine"
+          - php-version: "8.3"
+            php-image: "public.ecr.aws/docker/library/php:8.3-cli-alpine"
+
+    steps:
+      - name: "Install OS dependencies"
+        run: apk add --no-cache bash git su-exec
+
+      - name: "Install composer"
+        run: curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/bin --filename=composer --version=2.6.5
```
Comment (marmichalski) [inline .github/workflows/ci.yaml:44]:
```suggestion
        run: curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/bin --filename=composer --version=2.7.1
```

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+          key: ${{ runner.OS }}-${{ matrix.php-version }}-static-analysis-${{ hashFiles('**/composer.lock') }}-${{ github.run_id }}-${{ github.run_number }}
+          restore-keys: |
+            ${{ runner.OS }}-${{ matrix.php-version }}-static-analysis-${{ hashFiles('**/composer.lock') }}-${{ github.run_id }}
+            ${{ runner.OS }}-${{ matrix.php-version }}-static-analysis-${{ hashFiles('**/composer.lock') }}-
+            ${{ runner.OS }}-${{ matrix.php-version }}-static-analysis-
+
+      - name: "Ensure application writable for non-root user"
+        run: "chown -R 1000:1000 ."
+
+      - name: "Install composer dependencies"
```
Comment (marmichalski) [inline .github/workflows/ci.yaml:65]:
```suggestion
      - name: "Update composer platform version"
        if: ${{ matrix.dependencies != 'locked' && matrix.php-version != '8.1' }}
        run: "su-exec 1000 composer config platform.php ${{ matrix.php-version }}"

      - name: "Install composer dependencies"
```

File: src/CorrelationId.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+declare(strict_types=1);
+
+namespace Lendable\Message;
+
+use Ramsey\Uuid\Uuid as RamseyUuid;
+use Ramsey\Uuid\UuidInterface;
+
+/**
+ * An identifier that crosses multiple commands/events.
```
Comment (marmichalski) [inline src/CorrelationId.php:11]:
```suggestion
 * An identifier that crosses multiple messages.
```

💡

File: src/CorrelationId.php [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+ *
+ * This is generated at the initiation point of a flow
+ * and carried through to all subsequent commands and
+ * events to enable tracking of the flow.
```
Comment (marmichalski) [inline src/CorrelationId.php:15]:
```suggestion
 * and carried through to all subsequent messages to
 * enable tracking of the flow.
```

💡

**Review by marmichalski — APPROVED** [review]

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (ben-challis) [reply .github/workflows/ci.yaml:44]: I was just going to let Renovate pick this up immediately, but why not!

File: phpstan.neon [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  -
+    class: Ergebnis\PHPStan\Rules\Classes\NoExtendsRule
+    arguments:
+      classesAllowedToBeExtended:
+        - Exception
+        - RuntimeException
+        - Error
+        - LogicException
+        - InvalidArgumentException
+        - Symfony\Component\Console\Command\Command
```
Comment (ben-challis) [inline phpstan.neon:31]:
```suggestion
```

File: src/Uuid.php [hunk 6]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Lendable\Message;
+
+interface Uuid
```
Comment (ben-challis) [inline src/Uuid.php:7]: Feel we could drop this?

**Pull request merged**